### DISCUSSION TITLE
Question about how source data reader works in specific situation

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5918] [Q&A] [answered]
I have a data reader channel that calls a stored proc to pull back a group of patients. As the data is sent to the destination, there is a response transformer that goes back into the database and updates a flag so that patient won't get picked up again.

My question is, if the source is set to run every 15 minutes, and it takes longer than that to process all of the patients, will the source still poll again 15 minutes later, potentially picking up some of the same patients that it was currently processing?  I'm thinking that is the case.

**Comment by pacmano1** [comment]
No, it should not poll again if currently processing messages.

**Comment by rutledj** [comment]
Is that true even if the messages have processed through the source but are queued on the destination awaiting processing?

**Reply by jonbartels** [reply, marked as answer]
No it is not. If the source connector is "done" with a message then the poller will fire again. "done" in this case means that the source connector has transformed the message and passed to the destination queue or the source has enqueued the message in the source queue.

**Reply by rutledj** [reply]
Kinda what I expected.

**Reply by pacmano1** [reply]
Alas, @jonbartels hates source queues, but here is a another potential reason to use them.

**Reply by tonygermano** [reply]
No queuing is desired here if the intention is to wait for all of the records to be processed before the next poll begins. Queuing on the source or destination will make processing asynchronous and allow the next poll to run even if the records have not finished processing.

**Reply by tonygermano** [reply]
"done" actually means the post-processor has finished running (if not using source queue) and an upstream response has been sent if configured, which happens after the queuing events that @jonbartels mentioned. I would update the records in the post-processor instead of the response transformer if you intend to have destination queuing enabled and you only need to flag the records so they are not picked up again rather than waiting for them to finish processing.

The Database Reader has its own post-processor specifically for requirements such as these, but you can use the channel post-processor, too.
